src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.NamingStyle.cs

```diff
@@ -0,0 +1,94 @@
+            string namingRuleName,
+            IReadOnlyDictionary<string, object> allRawConventions,
+            out NamingStyle namingStyle)
+        {
+            namingStyle = null;
+            if (!TryGetNamingStyleTitle(namingRuleName, allRawConventions, out string namingStyleTitle))
+            {
+                return false;
+            }
+
+            string requiredPrefix = GetNamingRequiredPrefix(namingStyleTitle, allRawConventions);
+            string requiredSuffix = GetNamingRequiredSuffix(namingStyleTitle, allRawConventions);
+            string wordSeparator = GetNamingWordSeparator(namingStyleTitle, allRawConventions);
+            var capitalization = GetNamingCapitalization(namingStyleTitle, allRawConventions);
+
+            namingStyle = new NamingStyle()
+            {
+                Name = namingStyleTitle,
+                Prefix = requiredPrefix,
+                Suffix = requiredSuffix,
+                WordSeparator = wordSeparator,
+                CapitalizationScheme = capitalization
+            };
+
+            return true;
+        }
+
+        private static bool TryGetNamingStyleTitle(
+            string namingRuleName,
+            IReadOnlyDictionary<string, object> conventionsDictionary,
+            out string namingStyleName)
+        {
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_rule.{namingRuleName}.style", out object result))
+            {
+                namingStyleName = result as string;
+                return namingStyleName != null;
+            }
+
+            namingStyleName = null;
+            return false;
+        }
+
+        private static string GetNamingRequiredPrefix(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+            => GetStringFromConventionsDictionary(namingStyleName, "required_prefix", conventionsDictionary);
+
+        private static string GetNamingRequiredSuffix(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+            => GetStringFromConventionsDictionary(namingStyleName, "required_suffix", conventionsDictionary);
+
+        private static string GetNamingWordSeparator(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
+            => GetStringFromConventionsDictionary(namingStyleName, "word_separator", conventionsDictionary);
+
+        private static string GetStringFromConventionsDictionary(string namingStyleName, string optionName, IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_style.{namingStyleName}.{optionName}", out object result))
+            {
+                return result as string ?? string.Empty;
+            }
+
+            return string.Empty;
+        }
+
+        private static Capitalization GetNamingCapitalization(string namingStyleName, IReadOnlyDictionary<string, object> conventionsDictionary)
```

Comment:
I suspect this function is now just `ParseCapitalizationScheme(GetStringFromConventionsDictionary(...))`